consensus/broadcast/shardChainMessenger.go

```diff
@@ -123,6 +128,23 @@ func (scm *shardChainMessenger) BroadcastHeader(header data.HeaderHandler, pkByt
 	return nil
 }
 
+// BroadcastEquivalentProof will broadcast the proof for a header on the shard metachain common topic
+func (scm *shardChainMessenger) BroadcastEquivalentProof(proof *block.HeaderProof, pkBytes []byte) error {
+	if check.IfNilReflect(proof) {
+		return spos.ErrNilHeader
```

Comment:
same here

Reply:
maybe we can have a common inner function here for shard and meta and pass the topic

Reply:
fixed

Reply:
applied the suggestion